src/main/java/remer/sudoku/SudokuController.java

```diff
@@ -0,0 +1,30 @@
+package remer.sudoku;
+
+import java.util.List;
+
+public class SudokuController
+{
+    private final SudokuGui gui;
+    private Sudoku sudoku;
+
+    public SudokuController(SudokuGui gui, int[][] initialBoard)
+    {
+        this.gui = gui;
+        this.sudoku = new Sudoku(initialBoard);
+    }
+
+    // Method to validate the current board and update the GUI
+    public void validateBoard()
+    {
+        int[][] board = gui.getCurrentBoard();
```

Comment:
You pass a 2d array in the constructor. Why are you also calling get current board which returns a 2d array?